app/src/main/java/org/oppia/app/player/state/itemviewmodel/RatioExpressionInputInteractionViewModel.kt

```diff
@@ -0,0 +1,115 @@
+package org.oppia.app.player.state.itemviewmodel
+
+import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.databinding.Observable
+import androidx.databinding.ObservableField
+import org.oppia.app.R
+import org.oppia.app.model.AccessibleAnswer
+import org.oppia.app.model.Interaction
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.UserAnswer
+import org.oppia.app.parser.StringToRatioParser
+import org.oppia.app.player.state.answerhandling.AnswerErrorCategory
+import org.oppia.app.player.state.answerhandling.InteractionAnswerErrorOrAvailabilityCheckReceiver
+import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
+import org.oppia.domain.util.toAccessibleAnswerString
+import org.oppia.domain.util.toAnswerString
+
+/** [StateItemViewModel] for the ratio express input interaction. */
+class RatioExpressionInputInteractionViewModel(
+  interaction: Interaction,
+  private val context: Context,
+  val hasConversationView: Boolean,
+  val isSplitView: Boolean,
+  private val interactionAnswerErrorOrAvailabilityCheckReceiver: InteractionAnswerErrorOrAvailabilityCheckReceiver // ktlint-disable max-line-length
```

Comment:
Prefer not disabling ktlint checks--rename the variable to make it fit.

Reply:
done